platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -122,6 +127,25 @@ def get_name(self):
         return self._name
 
 
+    def _read_vpd_file(self, filename):
+        """
+        Read a vpd file parsed from eeprom with keys and values.
+        Returns a dictionary.
+        """
+        result = {}
+        try:
+            if not os.path.exists(filename):
+                return result
+            with open(filename, 'r') as fileobj:
+                for line in fileobj.readlines():
+                    key = line.split(":")[0].strip()
+                    val = line.split(":")[1].strip()
+                    result[key] = val
+        except Exception as e:
+            logger.log_info("Fail to read file {} due to {}".format(filename, repr(e)))
```

Comment:
should the log level be info or error?

Reply:
Not sure, right now it fails "gracefully" if the sysfs isn't available (i.e. the hardware isn't installed or the driver isn't present). If this is not the case then I can make it an error.

Reply:
I would suggest changing the log level to error because this is an unexpected problematic situation.